api/buckets/service.go

```diff
@@ -817,30 +997,38 @@ func (s *Service) nodeToItem(ctx context.Context, node ipld.Node, pth string, ke
 	if err != nil {
 		return nil, err
 	}
-	item := &pb.ListPathItem{
-		Cid:  node.Cid().String(),
-		Name: gopath.Base(pth),
-		Path: pth,
-		Size: int64(stat.CumulativeSize),
+	item := &pb.PathItem{
+		Cid:      node.Cid().String(),
+		Name:     gopath.Base(pth),
+		Path:     pth,
+		Size:     int64(stat.CumulativeSize),
+		Metadata: pmd,
 	}
-	for _, l := range node.Links() {
-		if l.Name == "" {
-			break
+	if pn, ok := node.(*dag.ProtoNode); ok {
```

Comment:
This is a better way to determine if this node is a directory. The prior method could not detect empty dirs (they were marked as non a directory).